packages/docusaurus-mdx-loader/src/remark/transformImage/index.js

```diff
@@ -13,17 +13,13 @@ const escapeHtml = require('escape-html');
 const {getFileLoaderUtils} = require('@docusaurus/core/lib/webpack/utils');
 const {posixPath, toMessageRelativeFilePath} = require('@docusaurus/utils');
 
-const {
-  loaders: {inlineMarkdownImageFileLoader},
-} = getFileLoaderUtils();
+const {assetQuery} = getFileLoaderUtils();
 
 const createJSX = (node, pathUrl) => {
   const jsxNode = node;
   jsxNode.type = 'jsx';
   jsxNode.value = `<img ${node.alt ? `alt={"${escapeHtml(node.alt)}"} ` : ''}${
-    node.url
-      ? `src={require("${inlineMarkdownImageFileLoader}${pathUrl}").default}`
-      : ''
+    node.url ? `src={require("${pathUrl}?${assetQuery}")}` : ''
```

Comment:
I recommend to avoid this stuff, you can do `src={new URL('./image.png', import.meta.url)}`, it is more native for ES modules and in future we should prefer to this way to get asset URL

Reply:
Thanks, replaced and it seems to work as before

Is it planned to support some kind of ESM-based DX? (I mean similar to Snowpack?)

Is it the recommended way to link to a file now?

```jsx
<a
  target="_blank"
  href={new URL(
    '../../assets/docusaurus-asset-example-pdf.pdf',
    import.meta.url,
  ).toString()}>
  Download this PDF
</a>
```

Reply:
> Is it planned to support some kind of ESM-based DX? (I mean similar to Snowpack?)

We thought about it and measured it, but at the moment memory cache (default in development mode) is faster and more stable, try to get snowpack and build big project and watch it, most likely you won't even be able to run it, with 5k modules my browser just freeze, with 10k snowpack fails.

In only JS world it is easier to implement ESM in browser, because you need parse `import`/`export` tokens and build graph and watch them, but when you get ts/css/assets/less/sass/pofyfills/etc and etc it is not easy, you need to write tokenizer for each source and regexps is not safe solution, other problems source maps (here long story). 

There is also a problem that dev and prod will be different, it can lead to unexpected problems too. 

And here a lot of limitations.

I don't want to say it is impossible, I want to say we need some more time for this.

> Is it the recommended way to link to a file now?

Yes, we don't have `import file from './text.pdf'` in spec, so it is recommend way, maybe `import assert` will change something in future https://github.com/tc39/proposal-import-assertions